thread/thread.cpp

```diff
@@ -1319,9 +1325,14 @@ R"(
     {
         if (unlikely(!th))
             LOG_ERROR_RETURN(EINVAL, , "invalid parameter");
-        if (unlikely(th->state != states::SLEEPING)) return;
+        if (unlikely(th->state != states::SLEEPING)) {
+        out: // may have thread_yield()-ed
+            if (th->state != states::READY || th->error_number == 0)
```

Comment:
Should it be
`th->state == states::READY && th->error_number == 0`?
Means thread is in READY queue and error number not set